intel-retail/retail-use-cases
# feat: make helm service for capi yolov8 ensemble

## PR Checklist



## What are you changing?

- add minikube -based implementation for C-API yolov8 ensemble
- the detailed documentation PR is here: https://github.com/intel-retail/documentation/pull/36  

## Issue this PR will close

close: **#43**

## Anything the reviewer should know when reviewing this PR?

## Test Instructions if applicable

- git clone this PR
- you will need minikube to test this PR alone with kubectl command as this PR is to run on helm/kubernetes
- change directory to `retail-use-cases/use-cases/get_capi/helm`
- on console run `make minikube_build_capi_yolov8_ensemble` to build the docker image into minikube and you should see build successfully
- then run `make run_capi_yolov8_ensemble` and you should see it runs up
- use `kubectl get pod -A` to check all pods are running ok.
- wait for a while, then you can check the fps result logs using command `make minikube_pod_log` and you should see something like this:
```console
...
Avg. Pipeline Throughput FPS: 30.000000
Avg. Pipeline Latency (ms): 46
Max. Pipeline Latency (ms): 49
Min. Pipeline Latency (ms): 45
Connected to MQTT broker: tcp://mqtt-broker:1883
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 30.000000
waiting for up to 10 seconds for publishing...
delivered token: 1 status code = 0
3 object(s) detected at 2024-10-16.22:40:19
Avg. Pipeline Throughput FPS: 20.000000
Avg. Pipeline Latency (ms): 48
Max. Pipeline Latency (ms): 56
Min. Pipeline Latency (ms): 45
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 20.000000
waiting for up to 10 seconds for publishing...
delivered token: 2 status code = 0
4 object(s) detected at 2024-10-16.22:40:21
Avg. Pipeline Throughput FPS: 22.500000
Avg. Pipeline Latency (ms): 52
Max. Pipeline Latency (ms): 68
Min. Pipeline Latency (ms): 45
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 22.500000
waiting for up to 10 seconds for publishing...
delivered token: 3 status code = 0
2 object(s) detected at 2024-10-16.22:40:23
Avg. Pipeline Throughput FPS: 20.000000
Avg. Pipeline Latency (ms): 58
Max. Pipeline Latency (ms): 74
Min. Pipeline Latency (ms): 35
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 20.000000
waiting for up to 10 seconds for publishing...
delivered token: 4 status code = 0
2 object(s) detected at 2024-10-16.22:40:24
Avg. Pipeline Throughput FPS: 18.750000
Avg. Pipeline Latency (ms): 45
Max. Pipeline Latency (ms): 54
Min. Pipeline Latency (ms): 40
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 18.750000
waiting for up to 10 seconds for publishing...
delivered token: 5 status code = 0
2 object(s) detected at 2024-10-16.22:40:26
Avg. Pipeline Throughput FPS: 20.000000
Avg. Pipeline Latency (ms): 45
Max. Pipeline Latency (ms): 50
Min. Pipeline Latency (ms): 43
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 20.000000
waiting for up to 10 seconds for publishing...
delivered token: 6 status code = 0
3 object(s) detected at 2024-10-16.22:40:28
Avg. Pipeline Throughput FPS: 19.090910
Avg. Pipeline Latency (ms): 48
Max. Pipeline Latency (ms): 62
Min. Pipeline Latency (ms): 41
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 19.090910
waiting for up to 10 seconds for publishing...
delivered token: 7 status code = 0
3 object(s) detected at 2024-10-16.22:40:29
Avg. Pipeline Throughput FPS: 20.000000
Avg. Pipeline Latency (ms): 52
Max. Pipeline Latency (ms): 62
Min. Pipeline Latency (ms): 44
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 20.000000
waiting for up to 10 seconds for publishing...
delivered token: 8 status code = 0
3 object(s) detected at 2024-10-16.22:40:31
Avg. Pipeline Throughput FPS: 19.285715
Avg. Pipeline Latency (ms): 49
Max. Pipeline Latency (ms): 63
Min. Pipeline Latency (ms): 44
publishing messages to MQTT broker: tcp://mqtt-broker:1883
topic: capiyolov8ensemble/FPS messages: 19.285715
waiting for up to 10 seconds for publishing...
delivered token: 9 status code = 0

```

- to shutdown containers, run `make down_capi_yolov8_ensemble`

## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/retail-use-cases )
